File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/scttests/EntryReactionActionTest.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.scttests;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.entryreactionaction.EntryReactionActionStatemachine;
import org.yakindu.scr.entryreactionaction.EntryReactionActionStatemachine.State;

/**
 * Unit TestCase for EntryReactionAction
 */
@SuppressWarnings("all")
public class EntryReactionActionTest {
	private EntryReactionActionStatemachine statemachine;	
	
	@Before
	public void setUp() {
		statemachine = new EntryReactionActionStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}
	
	@Test
	public void testentryTransitionActionOnStatechartEnter() {
		statemachine.enter();
		assertTrue(statemachine.getEnteredR1());
		 
		assertTrue(statemachine.getEnteredR2());
		 
		assertTrue(statemachine.getEnteredBdefault());
		 
		assertTrue(!statemachine.getEnteredBother());
		 
	}
	@Test
	public void testentryOnRTS() {
		statemachine.enter();
		statemachine.raiseB();
		statemachine.runCycle();
		statemachine.raiseD();
		statemachine.runCycle();
		statemachine.setEnteredR1(false);
		statemachine.setEnteredR2(false);
		statemachine.setEnteredBdefault(false);
		statemachine.setEnteredBother(false);
		statemachine.raiseB();
		statemachine.runCycle();
		assertTrue(!statemachine.getEnteredR1());
		 
		assertTrue(!statemachine.getEnteredR2());
		 
		assertTrue(!statemachine.getEnteredBdefault());
		 
		assertTrue(statemachine.getEnteredBother());
		 
	}
	@Test
	public void testnoEntryTransitionActionOnHistory() {
		statemachine.enter();
		statemachine.raiseB();
		statemachine.runCycle();
		statemachine.raiseD();
		statemachine.runCycle();
		statemachine.setEnteredR1(false);
		statemachine.setEnteredR2(false);
		statemachine.setEnteredBdefault(false);
		statemachine.setEnteredBother(false);
		statemachine.raiseD();
		statemachine.runCycle();
		assertTrue(!statemachine.getEnteredR1());
		 
		assertTrue(!statemachine.getEnteredR2());
		 
		assertTrue(!statemachine.getEnteredBdefault());
		 
		assertTrue(!statemachine.getEnteredBother());
		 
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/EntryReactionAction/EntryReactionActionTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "EntryReactionAction.h"

static EntryReactionAction statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		entryReactionAction_init(&statechart);
	}
};

void init(){
	entryReactionAction_enter(&statechart);
	entryReactionActionIface_raise_b(&statechart);
	entryReactionAction_runCycle(&statechart);
	entryReactionActionIface_raise_d(&statechart);
	entryReactionAction_runCycle(&statechart);
	entryReactionActionIface_set_enteredR1(&statechart,false);
	entryReactionActionIface_set_enteredR2(&statechart,false);
	entryReactionActionIface_set_enteredBdefault(&statechart,false);
	entryReactionActionIface_set_enteredBother(&statechart,false);
}

TEST_F(StatemachineTest, entryTransitionActionOnStatechartEnter) {					
	entryReactionAction_enter(&statechart);
	EXPECT_TRUE(entryReactionActionIface_get_enteredR1(&statechart));
	EXPECT_TRUE(entryReactionActionIface_get_enteredR2(&statechart));
	EXPECT_TRUE(entryReactionActionIface_get_enteredBdefault(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBother(&statechart));
}
TEST_F(StatemachineTest, entryOnRTS) {					
	init();
	entryReactionActionIface_raise_b(&statechart);
	entryReactionAction_runCycle(&statechart);
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR1(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR2(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBdefault(&statechart));
	EXPECT_TRUE(entryReactionActionIface_get_enteredBother(&statechart));
}
TEST_F(StatemachineTest, noEntryTransitionActionOnHistory) {					
	init();
	entryReactionActionIface_raise_d(&statechart);
	entryReactionAction_runCycle(&statechart);
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR1(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR2(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBdefault(&statechart));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBother(&statechart));
}


